TombEngine/Objects/TR3/Vehicles/big_gun.cpp

```diff
@@ -111,36 +111,30 @@ namespace TEN::Entities::Vehicles
 
 	void BigGunFire(ItemInfo* bigGunItem, ItemInfo* laraItem)
 	{
-		auto* lara = GetLaraInfo(laraItem);
 		auto* bigGun = GetBigGunInfo(bigGunItem);
 
 		short itemNumber = CreateItem();
-		auto* projectileItem = &g_Level.Items[itemNumber];
-
 		if (itemNumber != NO_ITEM)
 		{
+			auto* projectileItem = &g_Level.Items[itemNumber];
 			projectileItem->ObjectNumber = ID_ROCKET;
 			projectileItem->RoomNumber = laraItem->RoomNumber;
-
-			auto pos = GetJointPosition(bigGunItem, 2, Vector3i(0, 0, CLICK(1))); // CLICK(1) or 520?
+			auto pos = GetJointPosition(bigGunItem, 2, Vector3i(0, 0, CLICK(1)));
 			projectileItem->Pose.Position = pos;
-
-			InitialiseItem(itemNumber);
-
-			projectileItem->Animation.Velocity.z = 16;
 			projectileItem->Pose.Orientation = EulerAngles(
 				-((bigGun->XOrientFrame - 32) * ANGLE(1.0f)),
 				bigGunItem->Pose.Orientation.y,
 				0
 			);
+			InitialiseItem(itemNumber);
+
+			projectileItem->Animation.Velocity.z = 16;
 			projectileItem->ItemFlags[0] = BGUN_FLAG_UP_DOWN;
+			projectileItem->HitPoints = 1000; // NOTE: Time before it explose, TR5 use it, if 0, it will explode by default.
 
 			AddActiveItem(itemNumber);
-
-			lara->LeftArm.GunSmoke = 32;
-
 			for (int i = 0; i < 5; i++)
-				TriggerGunSmoke(pos.x, pos.y, pos.z, 0, 0, 0, 1, LaraWeaponType::RocketLauncher, lara->LeftArm.GunSmoke);
+				TriggerGunSmoke(pos.x, pos.y, pos.z, 0, 0, 0, 1, LaraWeaponType::RocketLauncher, 32);
```

Comment:
what's 32?

Reply:
value assigned to lara->LeftArm.GunSmoke, which is clearly not needed to do the smoke timer :)

Reply:
I don't understand, can you elaborate? How it worked before if it referenced that `lara->LeftArm.GunSmoke` value directly?

Reply:
It just use 32 as value and assign it to lara gunsmoke and pass it as argument to triggergunsmoke, which is not recommanded

Reply:
After discussion on a discord, I suggest to rollback this line to use `LeftArm.GunSmoke` again.

Reply:
Done